util/set/set.go

```diff
@@ -14,16 +14,32 @@ func SetOf[T comparable](slice []T) Set[T] {
 	return s
 }
 
-// Add adds e to the set.
+// Clone returns a new set cloned from the elements in s.
+func Clone[T comparable](s Set[T]) Set[T] {
+	clone := make(Set[T])
```

Comment:
Since we know the exact size, we might as well allocate to it:

```go
clone := make(Set[T], len(s))
```

Reply:
maps.Clone

Reply:
The funny thing is the TODO I wrote myself in corp says to use maps.Clone here :]